File: apps/web-vite/src/lib/sentry.ts
import * as Sentry from '@sentry/react';

const dsn = import.meta.env.VITE_SENTRY_DSN as string | undefined;

export function initSentry() {
  if (!dsn) return;

  const environment =
    (import.meta.env.VITE_APP_ENV as string | undefined) || import.meta.env.MODE;
  const isProd = environment === 'production';

  Sentry.init({
    dsn,
    environment,
    sendDefaultPii: false,
    tracesSampleRate: isProd ? 0.1 : 1.0,
  });
}

export function setSentryUser(user: { id: string; email?: string; name?: string } | null) {
  if (!dsn) return;
  if (!user) {
    Sentry.setUser(null);
    return;
  }
  Sentry.setUser({ id: user.id, email: user.email, username: user.name });
}

export { Sentry };

File: apps/api-go/internal/observability/sentry.go
package observability

import (
	"log"
	"time"

	"github.com/dakwah-depok/aisi/apps/api-go/internal/config"
	"github.com/getsentry/sentry-go"
)

// InitSentry initializes Sentry when SENTRY_DSN is set. Returns whether Sentry is active.
func InitSentry(c config.Config) bool {
	if c.SentryDSN == "" {
		return false
	}
	err := sentry.Init(sentry.ClientOptions{
		Dsn:              c.SentryDSN,
		Environment:      c.Environment,
		TracesSampleRate: map[bool]float64{true: 0.1, false: 1.0}[c.Production()],
		SendDefaultPII:   false,
	})
	if err != nil {
		log.Printf("sentry: gagal init: %v", err)
		return false
	}
	log.Printf("sentry: aktif (env=%s)", c.Environment)
	return true
}

// FlushSentry waits for buffered events before process exit.
func FlushSentry() {
	sentry.Flush(2 * time.Second)
}
